### ISSUE TITLE
Add dataset upload + dataset-mode runs

### ISSUE THREAD

**Issue by weijianzhg** [tunedtensor/tuned-tensor-cli#12]
The API now supports starting fine-tuning runs against an uploaded JSONL dataset (with train/validation/test splits) instead of the inline-examples cap. Tracked in [weijianzhg/tuned-tensor#39](https://github.com/weijianzhg/tuned-tensor/pull/39). The CLI needs three new touchpoints to make this usable from the terminal.

### 1. \`tt datasets upload\`

Multipart \`POST /api/v1/datasets\` with the file body, dataset name, optional description.

\`\`\`
tt datasets upload ./customer-support.jsonl --name customer-support-v1 \
  [--description \"10k support transcripts\"]
\`\`\`

- Accepts \`.jsonl\` files only.
- Server-side cap is **25 MB**; reject locally before upload to give a fast error.
- Each row must be \`{\"input\": string, \"output\": string}\` — non-empty fields. Validate locally so users don't burn an upload round-trip on a malformed file (the server rejects per-row anyway, but a local check is faster and avoids creating an \`invalid\`-status row).
- Print the resulting \`dataset.id\`, \`row_count\`, \`file_size_bytes\`, and \`status\` (\`validated\` on success). On \`status: invalid\`, surface the first 10 entries from \`validation_errors\` and exit non-zero.

### 2. \`tt datasets list\` / \`tt datasets get\`

Already wired (the endpoints existed); just confirm the table now includes:

- \`row_count\`
- \`file_size_bytes\` (formatted, e.g. \`3.2 MB\`)
- \`status\` — color/dim invalid rows

### 3. \`tt runs start\` — new flags

\`\`\`
tt runs start <spec-id> \
  [--dataset <dataset-id>] \
  [--split 80/10/10]   # train/val/test percentages, must sum to 100
  [--epochs 4] [--lora-rank 8] [--lora-alpha 16]
\`\`\`

When \`--dataset\` is provided:

- Pass \`dataset_id\` and \`split_ratios\` (decimals, e.g. \`{\"train\": 0.8, \"validation\": 0.1, \"test\": 0.1}\`) to \`POST /api/v1/behavior-specs/<spec-id>/runs\`.
- \`split_ratios\` is optional — server defaults to \`{0.8, 0.1, 0.1}\`. Only send it when \`--split\` is set.
- Reject \`--augment\` combined with \`--dataset\` (the server force-disables augmentation in dataset mode anyway, but mention it in the help text).

Without \`--dataset\` the existing inline-examples behavior is unchanged.

## API reference

- \`POST /api/v1/datasets\` — multipart, fields: \`file\`, \`name\`, \`description?\`. Returns dataset row.
- \`GET /api/v1/datasets?per_page=...\` — paginated list, user-scoped.
- \`POST /api/v1/behavior-specs/{spec_id}/runs\` — JSON body. New optional fields:
  - \`dataset_id: string (uuid)\`
  - \`split_ratios: { train: number, validation: number, test: number }\` — must sum to 1.0; train must be > 0.

## Docs

Update \`src/app/docs/datasets/page.tsx\` and \`src/app/docs/runs/page.tsx\` in the API repo to document the new flow once the CLI ships — but that's on the API side, not the CLI.

## References

- Server-side PR: [weijianzhg/tuned-tensor#39](https://github.com/weijianzhg/tuned-tensor/pull/39)
- Schema additions: \`splitRatiosSchema\`, \`startRunSchema.dataset_id\` ([api.ts](https://github.com/weijianzhg/tuned-tensor/blob/main/src/types/api.ts))
